### ISSUE TITLE
How to submit Coinbase String with the block? 

### ISSUE THREAD

**Issue by peysab** [coinfoundry/miningcore#564]
I am trying to add my pool (https://persianmine.com/pool/en/) to the list of known pools in the https://digiexplorer.info , I talked to the digibyte devs and they told me I need to submit the coinbase string along with the block so they can show my pool name as known pool in their explorer, and they gave me this link: http://learnmeabitcoin.com/glossary/coinbase-transaction 

now my question is does Miningcore supports this feature? and if yes  how should I set it up? and if no can you please tell me where in the code I should add it? @oliverw 

Thanks

**Comment by oliverw** [comment]
Try setting <code>coinbaseTxComment</code> for your coin in coins.json

**Comment by peysab** [comment]
ok will try thanks,
did you already set that for coinfoundry @oliverw ?

**Comment by peysab** [comment]
Hey @oliverw ,
I tried that and it leads to failing in submitting the blocks with following error:
` Submitting block 8294510 [000000000000000088347ca5288356881bb2f2ded841526539512646457ea112]

Mar 02 18:43:54 server.persianmine.com miningcore-app[13991]: [2019-03-02 18:43:54.8134] [W] [dgb] Block 8294510 submission failed with: invalid

Mar 02 18:43:54 server.persianmine.com miningcore-app[13991]: [2019-03-02 18:43:54.8134] [I] [NotificationService] Sending 'block submission failed' email to ********@gmail.com

Mar 02 18:43:54 server.persianmine.com miningcore-app[13991]: [2019-03-02 18:43:54.8134] [I] [dgb] [0HLKV35439IJ1] Share accepted: D=40574.327

Mar 02 18:43:54 server.persianmine.com miningcore-app[13991]: [2019-03-02 18:43:54.8218] [I] [dgb] [0HLKV35439IVA] Share accepted: D=56470.6

Mar 02 18:43:54 server.persianmine.com miningcore-app[13991]: [2019-03-02 18:43:54.8527] [E] [NotificationService] Error sending notification MailKit.Security.SslHandshakeException: An error occurred while attempting to establish an SSL or TLS connection.
`
Any ideas? and also any idea on the MailKit error?

**Comment by oliverw** [comment]
The coin has to support tx comments for that to work. YMMV.

**Comment by peysab** [comment]
I still couldn't  solve the issue @oliverw and I think it might be a good feature to promote your pool on the explorers. I asked one of the devs which was doing it with f2pool and here is what he replied:

```
this is how we did it for p2pool - 
After these lines in main.py start line 484
parser.add_argument('--merged',
        help='call getauxblock on this url to get work for merged mining (example: http://127.0.0.1:10332/)',
        type=str, action='append', default=[], dest='merged_urls')
Add this
parser.add_argument('--coinbtext',
        help='append this text to the coinbase',
        type=str, action='append', default=[], dest='coinb_texts')

And in work.py remove start line 318
     ] + ([mm_data] if mm_data else []) + [
                    ]) + self.current_work.value['coinbaseflags'])[:100],
And replace with
     ] + ([mm_data] if mm_data else []) + self.args.coinb_texts
                    ) + self.current_work.value['coinbaseflags'])[:100],


Now after doing this I am fairly sure that you need to start your p2pool with the arg --coinbtext 
example: pypy run_p2pool.py --net digibyte --coinbtext YOUR-COINB-STRING
```

I know we don't use python but I thought it might be helpful.
Any thoughts?
BTW the coin is DGB.

Thanks

**Comment by seperatis** [comment]
> I am trying to add my pool (https://persianmine.com/pool/en/) to the list of known pools in the https://digiexplorer.info , I talked to the digibyte devs and they told me I need to submit the coinbase string along with the block so they can show my pool name as known pool in their explorer, and they gave me this link: http://learnmeabitcoin.com/glossary/coinbase-transaction
> 
> now my question is does Miningcore supports this feature? and if yes how should I set it up? and if no can you please tell me where in the code I should add it? @oliverw
> 
> Thanks

Did you ever figure out how to change the coinbase string from Coinfoundry to something else?
